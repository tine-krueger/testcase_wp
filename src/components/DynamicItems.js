import PropTypes from 'prop-types'
import styled from 'styled-components/macro'
import useItems from '../hooks/useItems'
import ItemCard from './ItemCard'

DynamicItems.propTypes = {
    number: PropTypes.string
}

export default function DynamicItems(number) {
    const { queriedItems, isLoading, error } = useItems(number)

    return(
        <ItemsList>
            {
                isLoading ? 'Loading...' :
                error ? <p>{error}</p> : 
                queriedItems.map((item, i) => <ItemCard key={i} item={item} />)
            }
        </ItemsList>

    )
}

const ItemsList = styled.ul`
    list-style-type: none;
    padding: 0 1.5em 1em;
    margin-bottom: 0;

    @media (min-width: 768px) {
        padding: 0 2.5em 1em;
    }

    @media (min-width: 1200px) {
        padding: 0 10em 1em;
    }
`