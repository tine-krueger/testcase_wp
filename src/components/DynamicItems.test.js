import renderWithRouter from '../testSetup/setupTests'
import 'jest-styled-components'
import DynamicItems from './DynamicItems'

const mockAll = 'all'

describe ('Dynamic Items', () => {
    it('renders correctly', () => {
        const { container } = renderWithRouter(
            <DynamicItems number={mockAll} />
        )
        expect(container).toMatchSnapshot()

    })

})