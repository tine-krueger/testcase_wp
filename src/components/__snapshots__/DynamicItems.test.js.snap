// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dynamic Items renders correctly 1`] = `
.c0 {
  list-style-type: none;
  padding: 0 1.5em 1em;
  margin-bottom: 0;
}

@media (min-width:768px) {
  .c0 {
    padding: 0 2.5em 1em;
  }
}

@media (min-width:1200px) {
  .c0 {
    padding: 0 10em 1em;
  }
}

<div>
  <ul
    class="c0"
  >
    Loading...
  </ul>
</div>
`;
